mapit_gb/management/commands/mapit_UK_add_2019_structural_changes.py

```diff
@@ -0,0 +1,314 @@
+                        constructed.append(self.gillingham_ham_or_milton('S'))
+                    elif names == '‘Dewlands North’ and Dewlands South’ of the parish of Verwood':
+                        constructed.append(self.dewlands())
+            if len(qs) and constructed:
+                out = self._union(self.areas.filter(qs))
+                for g in constructed:
+                    out = out.union(g)
+            elif constructed:
+                out = constructed[0]
+                for g in constructed[1:]:
+                    out = out.union(g)
+            elif len(qs):
+                out = self._union(self.areas.filter(qs))
+            else:
+                raise Exception
+            self._create(new, 'UTW', out)
+
+    def gillingham_ham_or_milton(self, north_or_south):
+        polygons = self._intersect(
+            {'name': 'Gillingham', 'type__code': 'CPC', 'parent_area__in': self.districts},
+            {'name': 'Motcombe & Bourton', 'type__code': 'DIW', 'parent_area__in': self.districts})
+        polygons.sort(key=lambda x: x.extent[1])  # min_y
+        if north_or_south == 'N':
+            return polygons[1]
+        else:
+            return polygons[0]
+
+    def gillingham_rural(self):
+        return self._intersect(
+            {'name': 'Gillingham Rural', 'type__code': 'DIW', 'parent_area__in': self.districts},
+            {'name': 'Gillingham', 'type__code': 'CPC', 'parent_area__in': self.districts})
+
+    def symondsbury(self):
+        return self._intersect(
+            {'name': 'Bridport South', 'type__code': 'DIW', 'parent_area__in': self.districts},
+            {'name': 'Symondsbury', 'type__code': 'CPC', 'parent_area__in': self.districts})
+
+    def three_cross(self):
+        verwood_cpc = self.areas.get(name='Verwood', type__code='CPC', parent_area__in=self.districts)
+        verwood_ced = self.areas.get(name='Verwood', type__code='CED', parent_area=self.dorset)
+        verwood_ced_area = verwood_ced.polygons.all()[0].polygon
+        verwood_cpc_area = verwood_cpc.polygons.all()[0].polygon
+        return verwood_cpc_area.difference(verwood_ced_area)
+
+    def dewlands(self):
+        intersect = self._intersect(
+            {'name': 'Verwood', 'type__code': 'CPC', 'parent_area__in': self.districts},
+            {'name': 'Verwood', 'type__code': 'CED', 'parent_area': self.dorset})
+        verwood_east = self.areas.get(name='Verwood East', type__code='DIW', parent_area__in=self.districts)
+        verwood_area = verwood_east.polygons.all()[0].polygon
+        return intersect.difference(verwood_area)
+
+    # Only works for areas with one polygon
+    def _intersect(self, a1, a2):
+        a1 = self.areas.get(**a1).polygons.all()[0].polygon
+        a2 = self.areas.get(**a2).polygons.all()[0].polygon
+        intersect = a1.intersection(a2)
+        if intersect.geom_type == 'GeometryCollection':
+            intersect = [a for a in intersect if a.geom_type != 'LineString']
```

Comment:
Why are LineStrings being removed here?

Reply:
Intersecting two polygons that share a boundary, you end up with a nice polygon (or polygons in Symondsbury's case) of the intersection, but also loads of tiny LineStrings presumably caused by rounding errors, or something else I decided to just ignore :)

Reply:
That's fair enough!